src/env.c

```diff
@@ -1,6 +1,26 @@
 #include "rlang.h"
 
+
+bool r_is_unbound_value(SEXP x) {
+  return x == R_UnboundValue;
+}
+
 SEXP r_mut_env_parent(SEXP env, SEXP new_parent) {
```

Comment:
Could we use a typedef of `SEXP*` to indicate when an argument is modified?  It wouldn't improve the type safety, but it might improve documentation slightly